gslab_scons/tests/_side_effects.py

```diff
@@ -34,8 +34,8 @@ def side_effect(*args, **kwargs):
             source = match.group('source')
             log    = '%s.Rout' % re.sub('\.R', '', source)
     
-        if executable == "R CMD BATCH" and log:
-            with open(log.strip(), 'wb') as log_file:
+        if executable == "Rscript" and log:
```

Comment:
I'm not very familiar with `mock` either. But the basic idea is it does not actually run the command, but instead define what `side effect` this command will produce. Suppose we are running the following command from terminal:
`Rscript --no-save --no-restore --verbose script.R > script.log 2>&1`.

The effect of this command would be to run the script `script.R` and produce the log file `script.log`. When testing using `mock`, we do not actually run the command, but instead we define the behavior of this command in general (that is, to parse the command, get the executable, source, log, args, and do what the real command does). Then when tested, it works as if the command is run from the terminal. Does it make sense?